### PR TITLE
dockerTools: Fix chown in fakeRootCommands

### PR DESCRIPTION
**Pull request by the-sun-will-rise-tomorrow**
## Description of changes

`proot`'s `--root-id` "allows" `chown` only in the sense that it makes it succeed vacuously, i.e. a no-op. This is undesired if the goal is to actually create a layer with some files owned by different users.

Fortunately, `fakeroot` does allow persistence of emulated file owners, and it is possible to combine `fakeroot` with `proot`, so replace `proot --root-id` with `fakeroot` to do so.

This fixes a regression. Originally broken by 0aa8bd2dd97422b91ecb177cc93de1466be3da77 (#257462); #268458 fixed the build failure, but did not restore the original behavior.

CC @roberth @Mic92 @r-k-b

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/tests/docker-tools.nix [hunk 1]
```diff
@@ -46,6 +46,18 @@ let
...
     };
+
+  chownTestImage =
+    pkgs.dockerTools.streamLayeredImage {
+      name = "chown-test";
+      tag = "latest";
+      enableFakechroot = true;
+      fakeRootCommands = ''
+        touch testfile
+        chown 12345:12345 testfile
```
Comment (roberth) [inline nixos/tests/docker-tools.nix:57]:
```suggestion
        touch /testfile
        chown 12345:12345 /testfile
```

This way it matches the call in the `Entrypoint`.
It depends on `enableFakechroot = true;` but we have that.

File: nixos/tests/docker-tools.nix [hunk 2]
```diff
@@ -46,6 +46,18 @@ let
...
+      '';
+      config.Entrypoint = [ "${pkgs.coreutils}/bin/stat" "-c" "%u:%g" "/testfile" ];
```
Comment (roberth) [inline nixos/tests/docker-tools.nix:59]:
```suggestion
      config.Cmd = [ "${pkgs.coreutils}/bin/stat" "-c" "%u:%g" "/testfile" ];
```

I believe this is more idiomatic.

**Review by roberth — APPROVED** [review]

**Comment by WxNzEMof** [comment]
#283300 was merged first, so I updated this to only add a test.

CC @mplanchard @Mic92

**Comment by roberth** [comment]
@ofborg build nixosTests.dockerTools

**Comment by roberth** [comment]
Thank you!

**Pull request merged**